test/test_mcmc.py

```diff
@@ -40,19 +43,18 @@ def test_logistic_regression(algo):
     labels = dist.bernoulli(probs).rvs(random_state=random.PRNGKey(0))
 
     with validation_disabled():
-        def potential_fn(beta):
-            coefs_mean = np.zeros(dim)
-            coefs_lpdf = dist.norm(coefs_mean, np.ones(dim)).logpdf(beta)
-            logits = np.sum(beta * data, axis=-1)
-            y_lpdf = dist.bernoulli(logits, is_logits=True).logpmf(labels)
-            return - (np.sum(coefs_lpdf) + np.sum(y_lpdf))
+        def model(labels):
+            coefs = sample('coefs', dist.norm(np.zeros(dim), np.ones(dim)))
+            logits = np.sum(coefs * data, axis=-1)
+            return sample('obs', dist.bernoulli(logits, is_logits=True), obs=labels)
 
+        init_params, potential_fn = initialize_model(random.PRNGKey(2), model, (labels,), {})
         init_kernel, sample_kernel = hmc_kernel(potential_fn, algo=algo)
-        init_samples = np.zeros(dim)
-        hmc_state = init_kernel(init_samples,
+        hmc_state = init_kernel(init_params,
                                 step_size=0.1,
                                 num_steps=15,
                                 num_warmup_steps=warmup_steps)
-        hmc_states = lax.scan(lambda state, i: sample_kernel(state),
-                              hmc_state, np.arange(num_samples))
-        assert_allclose(np.mean(hmc_states.z, 0), true_coefs, atol=0.2)
+        sample_kernel = jit(sample_kernel)
```

Comment:
@fehiepsi - this brings down the time for HMC from 21s to 10s, but doesn't change it for NUTS, which I think already uses JAX primitives from the outset. Might be worth looking if we are missing out on similar opportunities for optimization.

Reply:
my bad, I think that the reason why I don't see performance different with jit+scan and non jit + scan is that I compiled jax with gpu recently. I just uninstalled it to do profiling.

Reply:
@neerajprad Could you make a gist to show the difference between jit and no jit for this test? In all of my tries, no-jit + scan is faster than jit + scan. I have observed the inconsistent behaviour of CPU vs GPU. I am curious if jax behaves differently for different CPU systems.

Reply:
Hmm..this is strange. I tried to recreate it but I am getting roughly the same time now (no-jit isn't faster though). I tried going back a few commits and I am still not able to see a difference, so I am guessing that it was just a noisy run earlier, or whatever was causing the overhead is resolved.